apps/ecommerce/frontend/src/components/ResourceCard.tsx

```diff
@@ -3,61 +3,74 @@ import { useState } from 'react';
+import { ExternalResource, ExternalResourceLink } from '../types';
+import { RenderDragFn } from '@contentful/field-editor-reference/dist/types';
+import getResourceProviderAndType from '../helpers/getResourceProviderAndType';
 
-const ResourceCard = (props: ResourceCardProps) => {
-  const debouncedValue = useDebounce(props.value, 300);
-  const { hydratedResourceData, isLoading, error, errorMessage, errorStatus } =
+interface Props {
+  value: ExternalResourceLink;
+  data?: ExternalResource;
+  index?: number;
+  total?: number;
+  onRemove: Function;
+  dragHandleRender?: RenderDragFn;
+  onMoveToTop?: Function;
+  onMoveToBottom?: Function;
+}
+
+const ResourceCard = (props: Props) => {
+  const { value, index, total, onRemove, dragHandleRender, onMoveToTop, onMoveToBottom } = props;
+
+  const debouncedValue = useDebounce(value, 300);
+  const { externalResource, isLoading, error, errorMessage, errorStatus } =
     useExternalResource(debouncedValue);
   const [showJson, setShowJson] = useState<boolean>(false);
 
   if (error) {
     return (
       <MissingResourceCard
-        index={props.index}
-        total={props.total}
-        onRemove={props.onRemove}
-        onMoveToBottom={props.onMoveToBottom}
-        onMoveToTop={props.onMoveToTop}
+        index={index}
+        total={total}
+        onRemove={onRemove}
+        onMoveToBottom={onMoveToBottom}
+        onMoveToTop={onMoveToTop}
         isLoading={isLoading}
         error={error}
-        value={JSON.stringify(props.value)}
+        value={JSON.stringify(value)}
         errorMessage={errorMessage}
         errorStatus={errorStatus}
-        dragHandleRender={props.dragHandleRender}
+        dragHandleRender={dragHandleRender}
       />
     );
   }
 
-  const [resourceProvider, resourceType] = props.value.sys?.linkType?.split(':');
+  const { resourceProvider, resourceType } = getResourceProviderAndType(value);
 
   return (
     <Card
       padding="none"
       isLoading={isLoading}
-      withDragHandle={!!props.dragHandleRender}
-      dragHandleRender={props.dragHandleRender}
+      withDragHandle={!!dragHandleRender}
+      dragHandleRender={dragHandleRender}
       isHovered={false}>
       <Box paddingLeft="spacingM" style={{ borderBottom: `1px solid ${tokens.gray200}` }}>
         <Flex alignItems="center" fullWidth={true} justifyContent="space-between">
           <Text fontColor="gray600" isWordBreak={true}>
             {resourceProvider} {resourceType}
           </Text>
           <Flex alignItems="center" isInline={true}>
-            {hydratedResourceData.status && (
-              <Badge variant="featured">{hydratedResourceData.status}</Badge>
-            )}
+            {externalResource.status && <Badge variant="featured">{externalResource.status}</Badge>}
             <ResourceCardMenu
-              onRemove={() => props.onRemove(props.index)}
+              onRemove={() => onRemove(index)}
```

Comment:
Note/Re-iteration since we found a bug: Being more declarative on how we set state rather than relying on some toggle method would improve code quality/bug prevention